### PR TITLE
Bug/slow delete namespace

### PR DESCRIPTION
**Pull request by ajfloeder**
Issue a format to each namespace before deleting it to ensure the delete operation is very fast.

I built the `tester` application which compiles the various nvme direct access functions for testing.
`cd internal/switchtec; env GOOS=linux GOARCH=amd64 GOPRIVATE=github.com go build -o tester main.go`

`tester` was then copied to rabbit-node-2 and used to format. I tested with both SED and non-SED drives to verify that the crypto erase option works.

```./tester format-ns 0x3b00@/dev/switchtec1 1```

Other testing:
In `nnf-sos`, vendored in new nnf-ec and tested in kind by creating and progressing a simple workflow.

### CODE REVIEW

File: pkg/manager-nvme/nvme_cli.go [hunk 1]
```diff
@@ -278,6 +278,51 @@ func (d *cliDevice) DeleteNamespace(namespaceId nvme.NamespaceIdentifier) error
...
+	// Example Command
+	//    # nvme format --force --namespace-id=1 /dev/nvme2
+	//    format: Success, formatted nsid:1
+
+	rsp, err := d.run(fmt.Sprintf("format %s --force --namespace-id=%d", d.dev(), namespaceID))
+	if err != nil {
+		return err
+	}
+
+	regex := regexp.MustCompile(`^format: Success, formatting nsid:(?P<NSID>\d+)\n`)
```
Comment (bdevcich-hpe) [inline pkg/manager-nvme/nvme_cli.go:291]: Should do this outside of the function so it only compiles once.

File: pkg/manager-nvme/nvme_cli.go [hunk 2]
```diff
@@ -278,6 +278,51 @@ func (d *cliDevice) DeleteNamespace(namespaceId nvme.NamespaceIdentifier) error
...
+	substrings := regex.FindStringSubmatch(rsp)
```
Comment (bdevcich-hpe) [inline pkg/manager-nvme/nvme_cli.go:292]: What is `format` outputting that we're using to search for a substring here? And why should it return 2 of them?

**Review by bdevcich-hpe — APPROVED** [review]

File: internal/switchtec/pkg/nvme/nvme.go [hunk 3]
```diff
@@ -644,6 +685,71 @@ func (dev *Device) DeleteNamespace(namespaceID uint32) error {
...
+
+	idns, err := dev.IdentifyNamespace(namespaceID, true /* namespace present */)
+	if err != nil {
+		return err
+	}
+
+	// If the drive supports crypto erase, use it.
+	formatOptions := FormatNs{
+		Format:             idns.FormattedLBASize.Format,
+		SecureEraseSetting: idctrl.Sanitize.CryptoErase,
```
[the remark this answers is not in the document]
Comment (ajfloeder) [reply internal/switchtec/pkg/nvme/nvme.go:706]: It sets the secure erase option based on the capability of the drive. For a self-encrypting drive, secure erase will be allowed, so we'll use crypto-erase to format. For a non-SED drive, we don't send the crypto-erase option.

File: pkg/manager-nvme/nvme_cli.go (hunk 1 above)
Comment (bdevcich-hpe) [reply pkg/manager-nvme/nvme_cli.go:291]: And I just noticed this is consistent with the other functions, so you can ignore this.

File: internal/switchtec/pkg/nvme/nvme.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajfloeder) [reply internal/switchtec/pkg/nvme/nvme.go]: @bdevcich-hpe I'm glad you pointed this out. The crypto-erase option is actually '2' vs '1' as it was originally coded.

File: pkg/manager-nvme/nvme_cli.go (hunk 2 above)
Comment (ajfloeder) [reply pkg/manager-nvme/nvme_cli.go:292]: We are trying to catch the substrings: 'Success formatting namespace:' and 'namespace-number' as substrings.
The 2nd should match the namespace-id that we input.

Comment (NateThornton) [reply pkg/manager-nvme/nvme_cli.go:292]: regex always returns the full match (if present) first, then the submatch[1] is the NSID

File: internal/switchtec/cmd/nvme/format_ns.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+// device to create a namespace with the given parameters. The next
+// available namespace ID is used for the create operation. Note that
+// create-ns does not attach the namespace to a controller, the attach-ns
+// command is needed.
+type FormatNsCmd struct {
+	Device      string `kong:"arg,required,type='existingFile',help='The nvme device or device over switchtec tunnel'"`
+	NamespaceID uint32 `kong:"arg,required,short='n',help='namespace to format'"`
+}
+
+// Run will run the Delete Namespace Command
```
Comment (NateThornton) [inline internal/switchtec/cmd/nvme/format_ns.go:38]: Update comment

**Review by NateThornton — APPROVED** [review]

**Pull request merged**